libraries/chain/include/eosio/chain/transaction.hpp

```diff
@@ -120,8 +120,10 @@ namespace eosio { namespace chain {
       vector<signature_type>    signatures;
       vector<bytes>             context_free_data; ///< for each context-free action, there is an entry here
 
-      const signature_type&     sign(const private_key_type& key, const chain_id_type& chain_id);
-      signature_type            sign(const private_key_type& key, const chain_id_type& chain_id)const;
+      const signature_type&     sign(const private_key_type& key, const chain_id_type& chain_id,
+                                     fc::require_canonical_t require_canonical = fc::require_canonical_t::yes);
```

Comment:
Can this not use a default value?

Reply:
restore default param in https://github.com/AntelopeIO/spring/pull/1708/commits/c1df814ba177648935a81701fca1574d8240d19f
because of usage in system-contracts repo.
```
/__w/spring/spring/system-contracts/tests/eosio.system_tester.hpp:1553:18: error: no matching function for call to 'eosio::chain::signed_transaction::sign(fc::crypto::private_key, eosio::chain::chain_id_type)'
2440
 1553 |          trx.sign( get_private_key( "producer1111"_n, "active" ), control->get_chain_id()  );
2441
      |          ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```